libraries/AP_Terrain/TerrainIO.cpp

```diff
@@ -178,8 +178,7 @@ void AP_Terrain::open_file(void)
     // create directory if need be
     if (!directory_created) {
         *p = 0;
-        mkdir(file_path, 0755);
```

Comment:
I wonder if perhaps NuttX doesn't correctly report back whether it's created a directory successfully or not?
I'm not qualified to review this part so maybe we can ask @tridge to say whether he knows anything about this.
If Tridge doesn't know then we can test that it does correct report back success or failure when creating a directory.